### ISSUE TITLE
[no-signal-compare?] new rule request to prevent comparing signals via < or >

### ISSUE THREAD

**Issue by skyleguy** [angular-eslint/angular-eslint#1753]
I would like to propose a new rule getting added to angular-eslint that will prevent two signals from being compared via <, <=, >, >= as it does not really make sense. After attending NgConf I noticed that it is very easy for even an expert angular developer to accidentally forgot to unbox two signals when trying to compare their unboxed values like so:

```
sig1 = signal(5)
sig2 = signal(2)

if (sig1 > sig2) // will never be true
if (sig1() > sig2()) // will be true in this case and is most likely always what we'd be going for in this scenario
```

it does seem valid to compare signals with === so it does seem like that should be allowed. 

I did open this request to @angular/core and was suggested that I open the feature request here instead. [ Here is the closed issue from @angular/core](https://github.com/angular/angular/issues/54986)

it seems it would be very helpful to get some warning/error feedback when attempts to do this are made and could possibly even have a quickFix option that just adds the calling of the signal instead of using the signal itself as experienced and new devs alike could spend plenty of time trying to figure out why their conditions are not being hit correctly when trying to use signals. The text could be similar to what we see when trying to compare numbers and strings?

![image](https://github.com/angular-eslint/angular-eslint/assets/14967057/bc9fcc98-bdd9-4a97-815e-f3b4926a5e0c)

**Comment by KevinKelchen** [comment]
I wonder if this could be expanded beyond comparison operators and apply to expressions in general. Expressions that evaluate to truthy/falsy would be common scenarios, but the scope would probably go beyond that.

A couple of examples:

This would flag as an issue:

```typescript
if (this.mySignal)
```

This would also flag as an issue:

```typescript
const myComputed = computed(() => this.mySignal + 1);
```

The rule could be disabled like other lint rules for the line that follows. This could be an opt-out for the [likely?] uncommon scenarios where we intend to check the signal reference itself instead of its unwrapped value.

**Comment by bjesuiter** [comment]
Yes, i have exactly this problem with the if statement. 

At our company we transfer a lot of components to signals but simply changing the inputs results in a lot of these cases, where the input is simply checkt for truthy/falsy in an if statement. 
Got some sample code from Stackoverflow:  https://stackoverflow.com/questions/78038046/how-to-prevent-angular-signal-references-being-used-instead-of-signal-values

```ts
public conditionalSignal = signal(false);
...

// Accidentally using the reference:
if (conditionalSignal) {
  ...
}

// Instead of something like this:
if (conditionalSignal()) {
  ...
}
```

This is really dangerous, as it's not easy to spot the wrong behavior while rewriting the components, especially when rewriting lots of components in a row. 

I'd really like to have an es-lint rule for that.

**Comment by ekrapfl** [comment]
I agree with expanding it to more places, similar to what was requested in #1891.

**Comment by MillerSvt** [comment]
I've update no-misused-signals proposal. I think this issue can be closed.

**Comment by maayanzrihann** [comment]
Should be added. can cause a lot bugs !

**Comment by JamesHenry** [comment]
I'll consolidate into #1891, thanks all
